sn_transfers/src/wallet/hot_wallet.rs

```diff
@@ -359,6 +359,13 @@ impl HotWallet {
             .collect()
     }
 
+    /// Checks whether the specified cash_note already presents
+    pub fn cash_note_presents(&mut self, id: &UniquePubkey) -> bool {
+        self.watchonly_wallet
+            .available_cash_notes()
+            .contains_key(id)
```

Comment:
I think we're using the mixture of both `Stateless` methods & `Stateful` methods in our payment forwarding + payment receiving code, and I think that might be prone to errors.

When we first receive the cashnote:
1. We call `cash_note_presents`, which returns the value from `watchonly_wallet` which is from memory inside `KeyLessWallet` (stateful in this context). 
2. We call `get_confirmed_spend` which reads the filesystem are returns the value (stateless).

When we forward the cashnote:
3. We call `remove_cash_notes_from_disk`, which removes the cashnote from the disk (stateless). Thus, it does not update the state of `KeyLessWallet` that is used during point-1
4. We call `remove_unconfirmed_spend_requests`, which writes the spend to `CONFIRMED_SPENDS_DIR_NAME` (stateless).


Thus, we read from fs directly in 3 places without updating our state, but return a value from the state at 1 place. 
And at point-1, `cash_note_presents` will always return false and will cause the same issue again.

I think we should move most of the logic to the Wallets, be it stateful or state-less, and use the same wallet to perform all the operations. I think that might be less error-prone and less confusing.

Reply:
Edit: `KeyLessWallet` is reloaded everytime we perform an OP. So the above effect is intended.